test/gurka/test_disable_hierarchy.cc

```diff
@@ -0,0 +1,83 @@
+#include "gurka.h"
+#include "proto/options.pb.h"
+#include <boost/format.hpp>
+#include <gtest/gtest.h>
+
+using namespace valhalla;
+
+// utils
+// Create costing options (reference: /test/astar.cc)
+void create_costing_options(Options& options, Costing::Type type) {
+  const rapidjson::Document doc;
+  sif::ParseCosting(doc, "/costing_options", options);
+  options.set_costing_type(type);
+}
+
+// Set disable_hierarchy_pruning to true in costing options
+void set_disable_hierarchy_pruning(Options& options, Costing::Type type) {
+  Costing* costing = &(*options.mutable_costings())[type];
+  auto* co = costing->mutable_options();
+  co->set_disable_hierarchy_pruning(true);
+}
+
+// Test the hierarchy limits are actually disabled when disable_hierarchy_pruning = true
+void doTest(Costing::Type costing_type) {
+  // Set costing options
+  Options options;
+  set_disable_hierarchy_pruning(options, costing_type);
+  create_costing_options(options, costing_type);
+  auto travel_mode = valhalla::sif::TravelMode::kDrive;
```

Comment:
no need to assign, as `CreateModeCosting` will do that for you
```suggestion
  auto travel_mode;
```